pkgs/tools/typesetting/tex/texlive/default.nix

```
@@ -121,36 +121,31 @@ let
       src = fetchurl { inherit urls sha512; };
 
       passthru = {
-        inherit pname tlType version;
-      } // lib.optionalAttrs (sha512 != "") { inherit src; };
+        inherit pname src tlType version;
+      };
       unpackCmd = file: ''
         tar -xf ${file} \
           '--strip-components=${toString stripPrefix}' \
           -C "$out" --anchored --exclude=tlpkg --keep-old-files
       '' + postUnpack;
 
-    in if sha512 == "" then
-      # hash stripped from pkgs.nix to save space -> fetch&unpack in a single step
+    in if fixedHash != null then
+      # hash already computed -> fetch&unpack with a single derivation to save space
       fetchurl {
-        inherit urls;
-        sha1 = if fixedHash == null then throw "TeX Live package ${tlName} is missing hash!"
-          else fixedHash;
-        name = tlName;
+        inherit passthru urls;
+        sha1 = fixedHash;
+        name = "texlive-${tlName}";
         recursiveHash = true;
         downloadToTemp = true;
         postFetch = ''mkdir "$out";'' + unpackCmd "$downloadedFile";
         # TODO: perhaps override preferHashedMirrors and allowSubstitutes
-     }
-        // passthru
+      }
 
     else runCommand "texlive-${tlName}"
-      ( { # lots of derivations, not meant to be cached
+      # hash yet to be computed -> build locally, do not cache
+      ( {
           preferLocalBuild = true; allowSubstitutes = false;
           inherit passthru;
```

```suggestion
          inherit passthru src;
```
And below `unpackCmd ''"${src}"''`

---

I meant to say
> And below `unpackCmd ''"$src"''`